### ISSUE TITLE
Gnome freezes caused by the indicator

### ISSUE THREAD

**Issue by Jarco** [philipphoffmann/gnome3-jenkins-indicator#83]
Hello,

Since I rebooted today gnome freezes for about half a second every time the indicator refreshes. 
Since I had it set to every 4.3 seconds you can imagine my system was quite unusable.

I don't really know what info to give you, but I did notice it spiked one cpu thread to 100% every time.
Since I disabled the extension my system is running well again.

**Comment by philipphoffmann** [comment]
Yeah, my believe is that the requests for updating the indicator are blocking the ui thread for some reason (usually thats happening if the list of jobs is very large). The only advice I have currently is to reduce the update interval.

If you find yourself able to fix this, I'd be happy to take a pull request
